desktop_server/src/com/mygdx/game/server/network/gameinstance/services/AttackEntityService.java

```diff
@@ -0,0 +1,46 @@
+package com.mygdx.game.server.network.gameinstance.services;
+
+import com.artemis.World;
+import com.mygdx.game.core.ecs.component.CanAttack;
+import com.mygdx.game.core.ecs.component.Coordinates;
+import com.mygdx.game.core.ecs.component.Stats;
+import lombok.extern.java.Log;
+
+import javax.inject.Inject;
+
+@Log
+public class AttackEntityService extends WorldService {
+
+  @Inject
+  public AttackEntityService() {
+    super();
+  }
+
+  public void attackEntity(int attacker, int attacked, World world) {
+    var coordinatesMapper = world.getMapper(Coordinates.class);
+    var canAttackMapper = world.getMapper(CanAttack.class);
+    if (!canAttackMapper.get(attacker).isCanAttack()
+        || !coordinatesMapper.get(attacker).equals(coordinatesMapper.get(attacked))) {
+      log.info("tried to attack unlawfully");
+      return;
+    }
+    var statsMapper = world.getMapper(Stats.class);
+    attackAndCounterAttack(statsMapper.get(attacker), statsMapper.get(attacked));
+    canAttackMapper.get(attacker).setCanAttack(false);
```

Comment:
Po zakończeniu tury np. w EndTurnService albo gdzieś w tej okolicy powinniśmy chyba zpowrotem ustawiać na true

Reply:
No tak, ale nie mamy jeszcze zdecydowanego chyba jak ma wyglądać czyszczenie takich rzeczy. To też chyba działoby się w tym samym miejscu co zwiększanie zasobów i budowanie jednostek. Dlatego teraz tego nie czyściłem.

Reply:
Okej, niech będzie